table/sstable/builder.go

```diff
@@ -285,8 +311,12 @@ func (b *Builder) finishBlock() error {
 		b.flushSingleKeyOldVers()
 	}
 	firstKey := b.tmpKeys.getEntry(0)
-	lastKey := b.tmpKeys.getLast()
-	blockCommonLen := keyDiffIdx(firstKey, lastKey)
+
+	// By @spongedu. disable key compressionj
+	// lastKey := b.tmpKeys.getLast()
+	// blockCommonLen := keyDiffIdx(firstKey, lastKey)
+	blockCommonLen := 0
```

Comment:
Why setting `blockCommonLen := 0` can disable key compression? 

I tried to run the code by setting it as 0 above, or having it as the official code, it looks like the `xxx.sst` file having a fixed size with official code but not with setting `blockCommonLen=0`

Perhaps we don't need to change here? The number of keys each sstable has should be the same and the file should be the same size or? 🤔

Reply:
> Why setting `blockCommonLen := 0` can disable key compression?
> 
> I tried to run the code by setting it as 0 above, or having it as the official code, it looks like the `xxx.sst` file having a fixed size with official code but not with setting `blockCommonLen=0`
> 
> Perhaps we don't need to change here? The number of keys each sstable has should be the same and the file should be the same size or? 🤔

See  `table/sstable/builder.go`, line `301`:

```
// entryFormat
// no old entry:
//  diffKeyLen(2) | diffKey | 0 | version(8) | value
// has old entry:
//  diffKeyLen(2) | diffKey | 1 | oldOffset(4) | version(8) | value
```

if `blockCommonLen` is. set to `0`, that means  `diffKeyLen` should be length of the key, and `diffKey` =`key`.  

But for now I do not check it yet.

I'll take a test here ~ . If it works ,maybe we can run `Greedy-PLR` on entries directly, not on blocks